rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/responder.hpp

```diff
@@ -250,9 +251,11 @@ class Responder
   noexcept
   {
     response.sequence_number_ = request_header.sequence_number;
-    response.client_guid_0_ = *(reinterpret_cast<const uint64_t *>(&request_header.writer_guid[0]));
-    response.client_guid_1_ = *(reinterpret_cast<const uint64_t *>(
-        &request_header.writer_guid[0] + sizeof(response.client_guid_0_)));
+    std::memcpy(
+      &response.client_guid_0_, &request_header.writer_guid[0], sizeof(response.client_guid_0_));
+    std::memcpy(
+      &response.client_guid_1_, &request_header.writer_guid[0] + sizeof(&response.client_guid_0_),
```

Comment:
Ah, that makes sense.  I hadn't checked the underlying types and just assumed that you were trying to get the 0th and 1st elements, not the 0th-7th and 8th-15th elements.